Automattic/wp-calypso
# Extract GuidedTour-related components from A4A

Related to 
- https://github.com/Automattic/dotcom-forge/issues/10138
- https://github.com/Automattic/wp-calypso/pull/97542
- https://github.com/Automattic/wp-calypso/pull/91100

## Proposed Changes

This PR extracts `GuidedTour`-related components from A4A and creates shared ones. 

## Why are these changes being made?


- https://github.com/Automattic/dotcom-forge/issues/10138
- Please refer to the Project Thread pbxlJb-6J5-p2 and the proposal post pfYzsZ-Y4-p2 for more context.

## Testing Instructions



Prerequisite 

* Go to Developer Console 
* POST to /me/preferences with the following data in `calypso_preferences`

```
{
  "site-management-panel": {"dismiss": false},
  "a4a-sites-add-new-site-tour-step-1": {"dismiss": false},
  "a4a-sites-add-new-site-tour-step-2": {"dismiss": false},
  "a4a-sites-tour": {"dismiss": false},
  "a4a-marketplace-tour": {"dismiss": false},
  "a4a-boost-agency-visibility-tour": {"dismiss": false},
  "a4a-start-referrals": {"dismiss": false},
  "a4a-invite-team": {"dismiss": false}
}
```

<img width="1432" alt="Screenshot 2024-12-19 at 13 00 01" src="https://github.com/user-attachments/assets/44203af4-48dc-4f3c-99cd-31f8b609419a" />

Calypso Live

* Go to /sites
* Select any site
* Observe the tour starts correctly 
* Refresh the page after the tour ends
* Observe the tour does not restart

A4A Live

* Go to /sites?tour=sites-walkthrough
	* Observe the tour starts correctly 
	* Refresh the page after the tour ends
	* Observe the tour does not restart
* Go to /sites?tour=add-new-site
	* Observe the tour starts correctly 
	* Refresh the page after the tour ends
	* Observe the tour does not restart

CC: @Automattic/a4a-genesis 

## Pre-merge Checklist
